src/ui/layout/qgslayoutmapgridwidgetbase.ui

```diff
@@ -957,31 +915,75 @@
             </property>
            </widget>
           </item>
-          <item row="4" column="0">
-           <widget class="QLabel" name="mAnnotationPositionLabelRight">
-            <property name="text">
-             <string>Right</string>
-            </property>
-           </widget>
+          <item row="8" column="1" colspan="2">
+           <widget class="QComboBox" name="mAnnotationPositionTopComboBox"/>
           </item>
-          <item row="20" column="0">
-           <widget class="QLabel" name="label_12">
-            <property name="text">
-             <string>Margin to map corner</string>
-            </property>
-           </widget>
+          <item row="6" column="1" colspan="2">
+           <widget class="QComboBox" name="mAnnotationDirectionComboBoxRight"/>
           </item>
-          <item row="20" column="1" colspan="2">
-           <widget class="QDoubleSpinBox" name="mRotatedAnnotationsMarginToCornerSpinBox">
-            <property name="toolTip">
-             <string>Outwards facing annotations closer to the corners than this margin will be ignored. This only makes sense for rotated grids.</string>
-            </property>
-            <property name="decimals">
-             <number>0</number>
+          <item row="17" column="0" colspan="3">
+           <widget class="QGroupBox" name="mRotatedAnnotationsGroupBox">
+            <property name="title">
+             <string>Rotate annotations</string>
             </property>
-            <property name="value">
-             <double>0.000000000000000</double>
+            <property name="checkable">
+             <bool>true</bool>
             </property>
+            <layout class="QFormLayout" name="formLayout_2">
+             <item row="0" column="0">
+              <widget class="QLabel" name="label_10">
+               <property name="text">
+                <string>Annotations alignment</string>
+               </property>
+              </widget>
+             </item>
+             <item row="0" column="1">
+              <widget class="QComboBox" name="mRotatedAnnotationsLengthModeComboBox"/>
+             </item>
+             <item row="1" column="0">
+              <widget class="QLabel" name="label_9">
+               <property name="text">
+                <string>Skip below angle</string>
```

Comment:
Can we also add tooltip, please? This label is not quite clear to me

Reply:
Ok